### PR TITLE
Simplify `F.deformable_convolution_2d_sampler` test

### PR DESCRIPTION
**Pull request by ishanrai05**
Simplifies `test_deformable_convolution_2d_sampler.py` using
`testing.FuntionTestCase`

Solves part of issue #6071

### CODE REVIEW

**Comment by ishanrai05** [comment]
I am getting an error: `AttributeError: 'SpatialTransformerSampler' object has no attribute 'st_desc'` on setting 'use_cuda` to `True`.
The absolute and relative tolerances also seem to be high.

**Comment by Crissman** [comment]
Runs ok on my system. Testing: Jenkins, test this please.

File: tests/chainer_tests/functions_tests/connection_tests/test_deformable_convolution_2d_sampler.py [hunk 1]
```diff
@@ -21,57 +16,80 @@
...
 
-        kh, kw, sy, sx, ph, pw = self.params
+        self.check_backward_options = {'atol': 1, 'rtol': 1}
 
-        self.stride = (sy, sx)
-        self.pad = (ph, pw)
+    def generate_inputs(self):
+        W = numpy.random.normal(
+            size=(self.out_channels, self.in_channels, self.kh, self.kw)
+            ).astype(self.dtype)
```
Comment (Crissman) [inline tests/chainer_tests/functions_tests/connection_tests/test_deformable_convolution_2d_sampler.py]:
```suggestion
        ).astype(self.dtype)
```

File: tests/chainer_tests/functions_tests/connection_tests/test_deformable_convolution_2d_sampler.py [hunk 2]
```diff
@@ -21,57 +16,80 @@
...
+        b = numpy.random.uniform(
+            size=(self.out_channels,)).astype(self.dtype)
 
-        self.W = numpy.random.normal(
-            size=(out_channels, in_channels, kh, kw)).astype(numpy.float32)
-        self.b = numpy.random.uniform(
-            size=(out_channels,)).astype(numpy.float32)
+        x = numpy.random.uniform(
+            size=(self.batch_size, self.in_channels, self.h, self.w)
+            ).astype(self.dtype)
```
Comment (Crissman) [inline tests/chainer_tests/functions_tests/connection_tests/test_deformable_convolution_2d_sampler.py]:
```suggestion
        ).astype(self.dtype)
```

File: tests/chainer_tests/functions_tests/connection_tests/test_deformable_convolution_2d_sampler.py [hunk 3]
```diff
@@ -21,57 +16,80 @@
...
 
-        self.x = numpy.random.uniform(
-            size=(batch_size, in_channels, h, w)).astype(numpy.float32)
+        offset = numpy.zeros(
+            (self.batch_size, 2 * self.kh * self.kw, self.out_h, self.out_w)
+            ).astype(self.dtype)
```
Comment (Crissman) [inline tests/chainer_tests/functions_tests/connection_tests/test_deformable_convolution_2d_sampler.py]:
```suggestion
        ).astype(self.dtype)
```

File: tests/chainer_tests/functions_tests/connection_tests/test_deformable_convolution_2d_sampler.py [hunk 4]
```diff
@@ -83,65 +101,84 @@ def test_forward_gpu(self):
...
+
+    def generate_inputs(self):
+        W = numpy.random.normal(
+            size=(self.out_channels, self.in_channels, self.kh, self.kw)
+            ).astype(self.dtype)
+        b = numpy.random.uniform(
+            size=(self.out_channels,)).astype(self.dtype)
+        x = numpy.random.uniform(
+            size=(self.batch_size, self.in_channels, self.h, self.w)
+            ).astype(self.dtype)
```
Comment (Crissman) [inline tests/chainer_tests/functions_tests/connection_tests/test_deformable_convolution_2d_sampler.py]:
```suggestion
        ).astype(self.dtype)
```

File: tests/chainer_tests/functions_tests/connection_tests/test_deformable_convolution_2d_sampler.py [hunk 5]
```diff
@@ -83,65 +101,84 @@ def test_forward_gpu(self):
...
+        self.out_w = utils.conv.get_conv_outsize(
+            self.w, self.kw, self.sx, self.pw)
+
+        self.check_backward_options = {'atol': 1, 'rtol': 5e-1}
+        self.check_forward_options = {'atol': 10, 'rtol': 5e-1}
+
+    def generate_inputs(self):
+        W = numpy.random.normal(
+            size=(self.out_channels, self.in_channels, self.kh, self.kw)
+            ).astype(self.dtype)
```
Comment (Crissman) [inline tests/chainer_tests/functions_tests/connection_tests/test_deformable_convolution_2d_sampler.py]:
```suggestion
        ).astype(self.dtype)
```

**Comment by ishanrai05** [comment]
@Crissman  I am still getting errors, I mentioned above.

**Comment by Crissman** [comment]
Hmm... Fine on my system. Let's see what Jenkins finds. Jenkins, test this please.

**Pull request closed without merging**